src/java_bytecode/ci_lazy_methods.cpp

```diff
@@ -322,8 +322,13 @@ void ci_lazy_methodst::initialize_all_needed_classes_from_pointer(
 {
   initialize_needed_classes_from_pointer(pointer_type, ns, needed_lazy_methods);
 
-  const pointer_typet &subbed_pointer_type=
-    pointer_type_selector.convert_pointer_type(pointer_type, ns);
+  // We pass an empty map to convert pointer type as we do not need to
+  // record the mapping of generic parameters to concrete types in the
+  // current context at this point (we are not initializing objects, just
+  // loading necessary classes). All the types that we need to load for
+  // future specialization must anyway appear on left-hand-side somewhere.
+  const pointer_typet &subbed_pointer_type =
```

Comment:
Note to check there is a test verifying this assumption

Reply:
I'm not sure how to check that the assumption is always true.. any idea what the test should have in it? I was trying to think of an example, something like this:
{code}
class A<T>
{
  public T a;
}

class B extends A<Integer>
{
}

public class base_test
{
  public B f;
  public A g = new A<Boolean>();

  public Integer test()
  {
    if(f.a==null && g.a)
    {
      return 0;
    }
    else if(f.a!=null)
    {
      return f.a;
    }
    else
    {
      return 1;
    }
  }
}
{code}
But this does not compile with an error:
{code}
error: bad operand types for binary operator '&&'      if(f.a==null && g.a)
{code}
Which supports the assumption.

Reply:
I guess I was just thinking:
```
public class base_test {
  public static String test(A<Integer> input) {
    if(input != null)
      return input.toString();
    else
      return "";
}
```
Then this should lazy load `Integer.toString()` 

(even better if it includes another test which is the same except the `input` type is `B` .

Reply:
I tested it, with `A<Integer> input` it loads the correct method, but not with `B input`. Will come up with a solution...

Reply:
I'm happy for this to be spun out into a separate ticket (with appropriate notes for QA) as this is still a big step forward, unless it seems super simple.

Reply:
It turns out it's not super simple :) I pulled the issue out into a separate ticket https://diffblue.atlassian.net/browse/TG-2664. The ticket contains the example from above.